Переделаны функции для включения, выключения и контроля состояния питания

DocenkoG opened the pull request:
Сделал отдельные процедуры для включения и для выключения модуля. А powerUpDown, судя по названию, непонятно что делает. Это вводит в заблуждение. Диаграммы для Sim900 приведены, например здесь http://www.geeetech.com/wiki/index.php/Arduino_GPRS_Shield
То есть, стандартная последовательность сигнала на PIN_PK применяется для переключения состояния. 
Для выключения предложил более интеллектуальный и быстрый алгоритм.
В папку примеров добавил скетч с использованием новых процедур.

DocenkoG committed:
Merge remote-tracking branch 'amperka/master'

DocenkoG committed:
Переделаны функции включения и выключения питания

При выключении, желательно не просто обесточить модуль, а выполнить
корректное отключение. Тогда отключение будет зарегистрировано в сети
мобильного оператора, а в энергонезависимый памяти сохранены текущие
режимы и переменные.
Одна и та же последовательность сигналов на управляющем спине приводит
как к выключению, так и включению модуля. Результат зависит от
предшествующего состояния.

DocenkoG committed:
Добавлена getImei, доработаны init и getDateTime

getDateTime - убрал секунды (не нужна такая избыточная точность). и
поменял тип возвращаемых данных.
int - добавил команду ATE0 - режим без эхо, чтоб легче было парсить
ответы FN команд. И изменил настройки CNMI на 2,0,2,1,1  (Это означает
- принимать СМС и записывать их в память, без сообщения о приходе СМС)
   Переформатирую потихоньку код к более компактному виду.

DocenkoG committed:
добавляю синхронизацию времени по NTP серверу

Не завершено, хотя работает правильно.
Надо сделать параметры и код завершения

nomad605dis commented:
Спасибо за вашу работу. В данный момент обновляем библиотеку и добавим ваши методы в неё.

DocenkoG committed:
Добавлена функция readBalance

Функция запрашивает текущий баланс и парит ответ мобильного оператора.
Результат возвращается в переменную типа int и в символьный буфер.

DocenkoG committed:
Неудачная доработка Gprs соединения

Стало виснуть на инициализации, Отваливается USB порт

DocenkoG committed:
Исправлены ошибки, добавлены функции до httpGet

DocenkoG committed:
Стабильная версия

Работа над библиотекой не завершена. Однако эта версия уже имеет
практическую ценность. Удалены ненужные, на мой взгляд методы.

DocenkoG committed:
Отладил функции, добавил примеров.

Протестировал и отладил добавленные функции. Написал массу примеров.
Ок.

DocenkoG committed:
http GET

Доработаны примеры и библиотека. Поверил на разных операторах в Москве:
МТС, Мегафон, Билайн. Работает удовлетворительно

dragrock commented:
Первая тестовая ветка
